### PR TITLE
feat(docker): create network req

### LINKED ISSUE #279: [Docker] Create network request
Create the structure to receive the `CreateNetowrkRequest` and publish the `AvailableNetworks` property.

<details><summary>Interfaces</summary>
<p>

### Create Network request

```json
{
  "interface_name": "io.edgehog.devicemanager.apps.CreateNetworkRequest",
  "version_major": 0,
  "version_minor": 1,
  "type": "datastream",
  "ownership": "server",
  "aggregation": "object",
  "mappings": [
    {
      "endpoint": "/network/id",
      "type": "string"
    },
    {
      "endpoint": "/network/driver",
      "type": "string"
    },
    {
      "endpoint": "/network/checkDuplicate",
      "type": "boolean"
    },
    {
      "endpoint": "/network/internal",
      "type": "boolean"
    },
    {
      "endpoint": "/network/enableIpv6",
      "type": "boolean"
    }
  ]
}
```

### Available Networks

```json
{
  "interface_name": "io.edgehog.devicemanager.apps.AvailableNetworks",
  "version_major": 0,
  "version_minor": 1,
  "type": "properties",
  "ownership": "device",
  "mappings": [
    {
      "endpoint": "/%{network_id}/id",
      "type": "string"
    },
    {
      "endpoint": "/%{network_id}/driver",
      "type": "string"
    },
    {
      "endpoint": "/%{network_id}/checkDuplicate",
      "type": "boolean"
    },
    {
      "endpoint": "/%{network_id}/internal",
      "type": "boolean"
    },
    {
      "endpoint": "/%{network_id}/enableIpv6",
      "type": "boolean"
    }
  ]
}
```

</p>
</details>

### PR DESCRIPTION
**Pull request by joshuachp**
Closes #279

### CODE REVIEW

File: edgehog-device-runtime-containers/src/docker/network.rs [hunk 1]
```diff
@@ -0,0 +1,483 @@
+// This file is part of Edgehog.
+//
+// Copyright 2023 SECO Mind Srl
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/docker/network.rs:3]:
```suggestion
// Copyright 2023-2024 SECO Mind Srl
```

File: edgehog-device-runtime-containers/src/docker/network.rs [hunk 2]
```diff
@@ -0,0 +1,483 @@
...
+        }
+    }
+
+    /// Set the id from docker.
+    fn update_id(&mut self, id: Option<String>)
+    where
+        S: Display,
+    {
+        if let Some(id) = id {
+            info!("using id {id} for network {}", self.name);
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/docker/network.rs:121]:
```suggestion
            debug!("using id {id} for network {}", self.name);
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: edgehog-device-runtime-containers/src/docker/network.rs [hunk 3]
```diff
@@ -0,0 +1,483 @@
...
+                Some(net) => {
+                    trace!("found network {net:?}");
+
+                    return Ok(false);
+                }
+                None => {
+                    debug!("network not found, creating it");
+                }
+            }
+        }
```
Comment (harlem88) [inline edgehog-device-runtime-containers/src/docker/network.rs:147]: You can use let-else and remove the match statement.
```suggestion
            if let  Some(net) = self.inspect(client).await? {
                    trace!("found network {net:?}");
                    return Ok(false);
                }else {
                    debug!("network not found, creating it");
              }
        }
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: edgehog-device-runtime-containers/src/docker/network.rs [hunk 4]
```diff
@@ -0,0 +1,483 @@
...
+/// Container network struct.
+///
+/// Networks are user-defined networks that containers can be attached to.
+#[derive(Debug, Clone, Eq)]
+pub struct Network<S> {
+    /// Id of the container network.
+    pub id: Option<String>,
+    /// The network's name.
+    pub name: S,
+    /// The network's name.
```
Comment (rgallor) [inline edgehog-device-runtime-containers/src/docker/network.rs:63]: comment since used above

**Review by rgallor — CHANGES_REQUESTED** [review]

**Comment by rgallor** [comment]
LGTM

File: edgehog-device-runtime-containers/src/docker/network.rs (hunk 3 above)
Comment (harlem88) [reply edgehog-device-runtime-containers/src/docker/network.rs]: this doesn't seem changed

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: edgehog-device-runtime-containers/src/docker/network.rs (hunk 3 above)
Comment (joshuachp) [reply edgehog-device-runtime-containers/src/docker/network.rs]: Mmm it did though, if you look at the log the comment is after the push. Maybe the page was outdated?

**Review by harlem88 — APPROVED** [review]

**Review by lucaato — APPROVED** [review]

**Pull request merged**